src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -349,6 +349,8 @@ public void clearReleasables(Lifetime lifetime) {
 
     public abstract FieldMapper smartNameFieldMapper(String name);
 
+    public abstract FieldMapper smartNameFieldMapperFromAnyType(String name);
```

Comment:
Can you add javadocs about how this is different than `smartNameFieldMapper`? I understand it tries to get it from the `mapperService` usually, but for instance, the percolator context throws an `UnsupportedOperationException` in the regular smartNameFieldMapper method, so I am not sure how they differ.